org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/contentmodel/XMLSchemaPublishDiagnosticsTest.java

```diff
@@ -108,14 +108,16 @@ public void schemaWithUrlInvalidPathWithNamespace() throws Exception {
 
 		String fileURI = "test.xml";
 		String xml = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\r\n" + //
-				"<invoice xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\r\n" + //
-				" xsi:schemaLocation=\"http://invoice.xsd\">\r\n" + //
-				"</invoice> \r\n" + //
-				"";
+				"<invoice xmlns=\"http://invoice.xsd\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\r\n" + //
+				" xsi:schemaLocation=\"http://invoice.xsd http://invoice.xsd http://invoice2.xsd\">\r\n" + //
+				"</invoice>";
 		XMLAssert.testPublishDiagnosticsFor(xml, fileURI, configuration, pd(fileURI, //
-				new Diagnostic(r(2, 20, 2, 40),
-				"SchemaLocation: schemaLocation value = 'http://invoice.xsd' must have even number of URI's.",
+				new Diagnostic(r(2, 20, 2, 79),

```

Comment:
Did the test fail before this change? If so, why? I'll also take a look.

Reply:
CI build is working. That's very strange.

Reply:
This test works, but the old test should also work (i.e. at least the "odd number of URIs" error should still be present).

Reply:
I see why this is implemented this way. A work around might be to move "xsi:schemaLocation even number of URI" validation into syntax validation?

Reply:
This kind of error is managed by Xerces when XSD validation is enabled, so I think it can be hard to do that. It's not a syntax problem but a XSD (syntax) problem.

I noticed that WTP XML validator disable XSD when:

 * the root element namespaceURI doesn't find the XSD to use in the xsi:schemaLocation(done in this PR)
 * and if the XSD is found, it check that XSD exists, in this case it disables the XSD validation.

I wonder if we should manage those 2 behaviors with a settings (ex : xml.validation.schema.disableWhenInvalid)

@fbricon what do you think about that?

Reply:
A new setting seems like it would be the best way forward.

Reply:
+1 for a new setting

Reply:
Should the setting be a part of this PR, or in a new PR?

Reply:
It should be part of this PR, but I need to find a good name for this settings. For the moment I though about `xml.validation.disableXSDValidationWhenNotFound`

Reply:
Today we have `xml.validation.schema` which is a boolean, I wonder if we could update this setting with 3 values:

* true
* false
* whenXSDUriIsValid

Reply:
@fbricon could you give us your opinion please about this settings.